Sources/YSwift/Protocols.swift

```diff
@@ -0,0 +1,66 @@
+import Yniffi
+
+/// A type that contains a reference to Yrs shared collection type.
+public protocol YCollection {
+    func sharedHandle() -> YrsSharedRef;
+}
+
+
+public protocol OriginProtocol {
```

Comment:
If you think there's a need or use case to compare to origins for equality, it might be worth specifying that OriginalProtocol conforms to : Equatable (and then add in the relevant methods to compare the different underlying types that could represent it). 

If this could be a concrete type, it might make some of the Swift side a bit easier, as the process of using Protocols as types (existential types in Swift) can get a bit weird/cranky with the compiler - at least if you ever need to dig into them or go beyond what's strictly defined as conformances on the protocol that you're using as the "wrapper". Beyond what's in this PR now, I don't see such a need, but figured it was worth mentioning.

Reply:
I have nothing to add for now apart from Joe has already mentioned. 

I especially agree with Joe on this one
> If this could be a concrete type, it might make some of the Swift side a bit easier

For sure it does makes sense to consider making it a concrete type if possible.

Reply:
I've modified it: now `Origin` is its own type wrapping uniffi code. It implements conversions from String and numeric types.